Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/DictionaryWorkspaceController.java

```diff
@@ -31,265 +30,200 @@
+    }
+
+    /**
+     * Retrieve all the dictionaries associated with workspace for deletion
+     * 
+     * @param workspaceId
+     * @param model
+     * @return Stirng - URL to redirect on success or failure
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.WORKSPACE, paramIndex = 1, userRole = {
+            RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/workspace/{workspaceid}/deletedictionary", method = RequestMethod.GET)
+    public String deleteWorkspaceDictionary(@PathVariable("workspaceid") String workspaceId, Model model,
+            Principal principal) throws QuadrigaStorageException, QuadrigaAccessException {
+        String userId = principal.getName();
+
+        List<IWorkspaceDictionary> dicitonaryList = null;
+        dicitonaryList = workspaceDictionaryManager.listWorkspaceDictionary(workspaceId, userId);
+        if (dicitonaryList == null) {
+            logger.info("Dictionar list is empty buddy");
+        }
+        model.addAttribute("dicitonaryList", dicitonaryList);
+        IWorkSpace workspace = wsManager.getWorkspaceDetails(workspaceId, userId);
+        model.addAttribute("workspacedetails", workspace);
+        model.addAttribute("workspaceId", workspaceId);
+        return "auth/workbench/workspace/deletedictionaries";
+    }
+
+    /**
+     * Delete the association of selected dictionaries from the workspace.
+     * 
+     * @param req
+     * @param workspaceId
+     * @param model
+     * @return String - URL to redirect the page on success or failure
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.WORKSPACE, paramIndex = 2, userRole = {
+            RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/workspace/{workspaceid}/deletedictionaries", method = RequestMethod.POST)
+    public String deleteWorkspaceDictionary(HttpServletRequest req, @PathVariable("workspaceid") String workspaceId,
+            Model model, RedirectAttributes attr, Principal principal)
+                    throws QuadrigaStorageException, QuadrigaAccessException {
+        String userId = principal.getName();
+        String msg = "";
+        int flag = 0;
+
+        String[] values = req.getParameterValues("selected");
+        if (values == null) {
+            attr.addFlashAttribute("show_error_alert", true);
+            attr.addFlashAttribute("error_alert_msg", "Please select a Dictionary");
+            return "redirect:/auth/workbench/workspace/" + workspaceId + "/deletedictionary";
+        }
+        for (int i = 0; i < values.length; i++) {
+            workspaceDictionaryManager.deleteWorkspaceDictionary(workspaceId, userId, values[i]);
+            if (!msg.equals("")) {

```

Comment:
msg is never set besides in line 198 if I'm not mistaken. Does deleteWorkspaceDictionary return a string to set msg? If not, this whole if is pointless. Please also check the other classes if the same thing is in there as well. I didn't notice it until now, but it's very likely I just overlooked it, they are all very similar.